### PR TITLE
Add compression in http res

### PR DESCRIPTION
**Pull request by dholms**
Add in compression middleware & explicitly tip it off that it should be compressing car files

Not implemented on repo subscription yet, we can discuss

Closes https://github.com/bluesky-social/atproto/issues/1007

### CODE REVIEW

File: packages/pds/package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "url": "https://github.com/bluesky-social/atproto.git",
     "directory": "packages/pds"
   },
-  "main": "src/index.ts",
+  "main": "dist/index.js",
```
Comment (bnewbold) [inline packages/pds/package.json:10]: intentional fix, or accidental commit?

Comment (Matrix89) [reply packages/pds/package.json:10]: What's even up with that, when I do `yarn install` is does it automagically

**Comment by bnewbold** [comment]
I dug through the docs of this package a bit, and we should verify if it is actually working.

It has a threshold of 1kbyte, below that it does not compress. Good! Only works if `Content-Length` is provided; should verify that is working for XRPC endpoints.

 It uses a MIME database to check if the payload is "compressible". Eg, `image/jpeg` won't be re-compressed (correct behavior). For CAR files, it isn't obvious if it will work or not. If not, one solution might be to just send a PR to the upstream project:
 
 https://www.npmjs.com/package/mime-db
 https://github.com/jshttp/mime-db/blob/master/src/custom-types.json

File: packages/pds/package.json (hunk 1 above)
Comment (dholms) [reply packages/pds/package.json:10]: accidental

yeah I'm honestly not sure where it comes from. has bitten us a couple times

**Comment by dholms** [comment]
ah yup it probably isn't working then 😅 

I threw this together in like 5 min just to get the ball rolling & didn't test it out at all yet

I'll give it some more love here soon

**Comment by bnewbold** [comment]
Was just poking at bsky.app page sizes, and getTimeline is like 110 KBytes+. Would be good to get this in even if it only works for big JSON responses!

**Comment by dholms** [comment]
@devinivy & I were just chatting about this!

gonna get back into it here soon

**Review by devinivy — APPROVED** [review]

**Review by devinivy — APPROVED** [review]
Ace!

**Pull request merged**